### PR TITLE
Using assertj as much as possible, adding some custom assertions  and fix some documentation.

### PR DESCRIPTION
**Pull request by mpawliszyn**

### CODE REVIEW

File: client-misk-embedded/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+POM_ARTIFACT_ID=client-misk
```
Comment (shellderp) [inline client-misk-embedded/gradle.properties]: should be -embedded

File: client-misk/build.gradle [hunk 2]
```diff
@@ -15,7 +15,8 @@ dependencies {
   implementation dep.kotlinStdLib
   implementation dep.wireMoshiAdapter
 
-  compile project(':client')
+  api project(':client')
+  testApi project(':client-misk-testing')
```
Comment (shellderp) [inline client-misk/build.gradle]: :+1:

File: client-misk/src/test/kotlin/app/cash/backfila/client/misk/hibernate/SingleInstanceHibernateBackfillTest.kt [hunk 3]
```diff
@@ -67,98 +49,105 @@ class UnshardedHibernateBackfillTest {
...
-    assertEquals(run.prepareBackfillResponse.instances.size, 1)
-    assertNull(run.instanceProgressSnapshot.values.single().keyRange.start)
-    // Trying to get a batch on an empty tablet gets nothing to execute.
-    assertTrue { run.singleScan().batches.isEmpty() }
-    assertTrue { run.complete() }
+    assertThat(run).hasSingleInstanceCursorThat().hasKeyRangeStartThat().isNull()
+    assertThat(run).hasSingleInstanceCursorThat().hasKeyRangeEndThat().isNull()
+    // Trying to scan for a batch on an empty tablet gets nothing to execute.
+    assertThat(run.singleScan().batches).isEmpty() // Check the returned scan
+    assertThat(run).isFinishedScanning()
```
Comment (shellderp) [inline client-misk/src/test/kotlin/app/cash/backfila/client/misk/hibernate/SingleInstanceHibernateBackfillTest.kt]: cool

File: client-misk-embedded/gradle.properties (hunk 1 above)
Comment (mpawliszyn) [reply client-misk-embedded/gradle.properties]: Yeah I thought I had deleted this... It is gone now.

File: client-misk/src/test/kotlin/app/cash/backfila/client/misk/hibernate/SingleInstanceHibernateBackfillTest.kt (hunk 3 above)
Comment (mpawliszyn) [reply client-misk/src/test/kotlin/app/cash/backfila/client/misk/hibernate/SingleInstanceHibernateBackfillTest.kt]: Glad you like it. Not sure I got everything 100% right but I think it's a step in the right direction.

I figure @swankjesse will either love it or hate it. I'm awaiting judgement. ⚖️

File: client-misk/build.gradle (hunk 2 above)
Comment (mpawliszyn) [reply client-misk/build.gradle]: Might as well fix one more.

**Comment by mpawliszyn** [comment]
@swankjesse made a good point that a lot of the methods I made were pretty similar to the way we access field in kotlin. So I cleaned it up a bit differently.

**Pull request merged**